boltup/boltup.sh

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+# 'boltup' script for installing and managing the bolt CLI tool.
+#
+# This script is a heavily inspired by 'foundryup' (https://github.com/foundry-rs/foundry/blob/master/foundryup/foundryup)
+# since it is a great way to install CLI tools. Kudos to the Foundry team :)
+
+BASE_DIR=${XDG_CONFIG_HOME:-$HOME}
+BOLT_DIR=${BOLT_DIR:-"$BASE_DIR/.bolt"}
+BOLT_BIN_DIR="$BOLT_DIR/bin"
+
+main() {
+  need_cmd git
+  need_cmd curl
+
+  while [[ -n $1 ]]; do
+    case $1 in
+      --)               shift; break;;
+
+      -t|--tag)     shift; BOLTUP_TAG=$1;;
+      --arch)       shift; BOLTUP_ARCH=$1;;
+      --platform)   shift; BOLTUP_PLATFORM=$1;;
+      -h|--help)
+        usage
+        exit 0
+        ;;
+      *)
+        warn "unknown option: $1"
+        usage
+        exit 1
+    esac; shift
+  done
```

Comment:
Can this be moved to a separate, documented function?
By documentation I mean something like:

"This function loops as long as $1 (the first positional argument) is not empty, and matches its value against flags. If a flag is found, the argument is discarded (via `shift`) and the value for the flag is read."

I know, I should RTFM but having gently reminders of bash syntax is nice.

Reply:
I didn't move it but I added the comment, as I don't want to introduce untested behaviour in a script that is battle tested (this is what foundry has been using for 2+ years)